### PR TITLE
Added Module 7 code for API Test automation using rest assured

### PR DESCRIPTION
**Pull request by raju1284**
Please review and provide your feedback

### CODE REVIEW

File: src/main/java/utilities/ServiceUtil.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+import io.restassured.response.Response;
+
+public class ServiceUtil {
+
+
+    public Response postResponseFromService(Object requestObject, String url, String projectName, ObjectMapper objectMapper, String token) throws JsonProcessingException {
+        return RestAssured.given().log().all()
+                .header("accept", "application/json")
+                .header("Content-Type", "application/json")
+                .header("Authorization", token)
```
Comment (KorolIvan) [inline src/main/java/utilities/ServiceUtil.java]: Take a look on the .headers(Map<Strint, Object>)
https://www.tutorialspoint.com/how-to-pass-more-than-one-header-in-a-request-in-rest-assured#:~:text=Passing%20them%20as%20a%20Map%20using%20the%20headers%20method.

You will use the same header set for all requests, I guess it make sense to have all of them in one place, and token you can pass as a parameter to the constructor. Is it make sense?

File: src/test/resources/Features/api/DashbaordAPI.feature [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    @Test2
+    Scenario Outline: Negative Test- Add existing Dashboard using API post request
+      When User creates the dashboard with  given dashboard name "<dashboardName>" and description "<dashboardDesc>" under the Project "<projectName>"
+      Then User receives "<statusCode>" status code
+      And User validates the error code "<errorCode>" and message "<message>"
+
+      Examples:
+        |dashboardName|dashboardDesc|projectName|statusCode|errorCode|message|
+        |apiTest6     |Test6        |Nagaraju_Dasari_Personal|409|4091|Resource 'apiTest6' already exists. You couldn't create the duplicate.|
+    @Test3
```
Comment (KorolIvan) [inline src/test/resources/Features/api/DashbaordAPI.feature:20]: Please split the test by new line,

File: src/test/resources/Features/api/DashbaordAPI.feature [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    @Test9
+    Scenario Outline: Negative Test - Add widget to the non Dashboard using API put request
+      When User adds the widget to the dashboardId "<dashboardId>" under the Project "<projectName>" with widget name "<widgetName>" Id "<widgetId>" and type "<widgetType>"
+      Then User receives "<statusCode>" status code
+      And User validates the error code "<errorCode>" and message "<message>"
+
+      Examples:
+        |dashboardId|projectName|statusCode|widgetName|widgetId|widgetType|errorCode|message|
+        |141905     |Nagaraju_Dasari_Personal|404|Widget1|142130|statisticTrend|40422|Dashboard with ID '141905' not found on project 'nagaraju_dasari_personal'. Did you use correct Dashboard ID?|
+    @Test10
```
Comment (KorolIvan) [inline src/test/resources/Features/api/DashbaordAPI.feature:84]: Ditto.

File: src/test/java/StepDefinition/api/DashboardAPITestSteps.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        sharedData.setResponse(sharedData.getServiceUtil().postResponseFromService(
+                sharedData.getCreateDashboardRequest(), fr.getPropertyValue("PostDashboard.Url"), projectName,sharedData.getObjectMapper(),fr.getPropertyValue("AuthorizationToken")));
+    }
+
+    @Then("User receives {string} status code")
+    public void validateTheResponseStatusCode(String statusCode)
+    {
+        Assert.assertEquals(statusCode,String.valueOf(sharedData.getResponse().statusCode()));
+    }
+    @And("User validates the error code {string} and message {string}")
```
Comment (KorolIvan) [inline src/test/java/StepDefinition/api/DashboardAPITestSteps.java:33]: New line is missed between methods.

File: src/test/java/StepDefinition/api/DashboardAPITestSteps.java [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+    }
+
+    @And("User validates the response name {string} and description {string}")
+    public void userValidatesTheDashboardData(String name,String description)
+    {
+        Assert.assertEquals(name,sharedData.getResponse().jsonPath().getString("name"));
+        Assert.assertEquals(description,sharedData.getResponse().jsonPath().getString("description"));
+    }
+    @When("User deletes the dashboard with given dashboard Id {string}  under the Project {string}")
```
Comment (KorolIvan) [inline src/test/java/StepDefinition/api/DashboardAPITestSteps.java:52]: Ditto.

File: src/test/java/StepDefinition/api/DashboardAPITestSteps.java [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+        sharedData.setAddWidgetRequest(AddWidgetRequest.builder()
+                .addWidget(sharedData.getAddWidget())
+                .build());
+
+        sharedData.setResponse(sharedData.getServiceUtil().putResponseFromAddWidgetToTheDashboard(
+                sharedData.getAddWidgetRequest(),fr.getPropertyValue("PutDashboard.Url"),projectName,dashboardId,sharedData.getObjectMapper(),fr.getPropertyValue("AuthorizationToken")));
+
+
+    }
+    @When("User deletes the widget in the dashboardId {string} under the Project {string} with widget Id {string}")
```
Comment (KorolIvan) [inline src/test/java/StepDefinition/api/DashboardAPITestSteps.java]: Ditto.

File: src/test/java/StepDefinition/api/DashboardAPITestSteps.java (hunk 4 above)
Comment (KorolIvan) [reply src/test/java/StepDefinition/api/DashboardAPITestSteps.java:33]: Please use the code formation with Ctr+Alt+L, to format the code as you have somewhere the Curly brackets in the different positions from method to method.

**Review by KorolIvan — COMMENTED** [review]
Please take a look on comments,

**Comment by raju1284** [comment]
Updated the code as per Review comments

**Review by KorolIvan — APPROVED** [review]
Please mark the left comments as resolved and u good to submit.
LGTM/Approve.

File: src/test/java/StepDefinition/api/DashboardAPITestSteps.java (hunk 4 above)
Comment (raju1284) [reply src/test/java/StepDefinition/api/DashboardAPITestSteps.java:33]: Resolved

File: src/test/resources/Features/api/DashbaordAPI.feature (hunk 3 above)
Comment (raju1284) [reply src/test/resources/Features/api/DashbaordAPI.feature:84]: Resolved

File: src/test/resources/Features/api/DashbaordAPI.feature (hunk 2 above)
Comment (raju1284) [reply src/test/resources/Features/api/DashbaordAPI.feature:20]: Resolved

File: src/main/java/utilities/ServiceUtil.java (hunk 1 above)
Comment (raju1284) [reply src/main/java/utilities/ServiceUtil.java]: Resolved

**Pull request merged**